src/main/scala/mesosphere/marathon/state/TaskFailure.scala

```diff
@@ -65,6 +66,26 @@ object TaskFailure {
       slaveId = if (proto.hasSlaveId) Some(proto.getSlaveId) else None
     )
 
+  object FromUnhealthyTaskKillEvent {
+    def unapply(event: UnhealthyTaskKillEvent): Option[TaskFailure] =
+      apply(event)
+
+    def apply(event: UnhealthyTaskKillEvent): Option[TaskFailure] = {
```

Comment:
I guess you use the same method signature as `FromMesosStatusUpdateEvent`.
Since there can be different status update this makes sense there, but we want to treat every kill event as task failure. 
Please remove the option from the apply method.